src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -133,8 +133,13 @@ void remove_redundant_reorders::run(program& p) {
         auto node = *itr++;
         if (!node->is_type<reorder>())  // only care for reorders
             continue;
-
         auto& r_node = node->as<reorder>();
+
+        // Do not opt out result reorder of Loop body network
+        bool is_loop_body_network_output = (r_node.get_program().is_body_program && r_node.is_output() && r_node.can_share_buffer());
+        if (is_loop_body_network_output)
+            continue;
```

Comment:
I think we should add addiitional condition as I mentioned before, 
1. is loob body output 
2. is backedge (from) and the the target node (to) is sharing memory

Reply:
Also, can_share_buffer is not relevant to the expected change.